features/step_definitions/menu_steps.rb

```diff
@@ -0,0 +1,18 @@
+Given("the following products exist:") do |table|
+  table.hashes.each do |product|
+    @menu = Menu.find_or_create_by(name: product[:Menu])
```

Comment:
Please don't use capital letters in hash keys. Not a good practice. 
Also, `@menu` is not used outside of this method. A local variable would do.